src/components/course/enrollment/hooks.js

```diff
@@ -0,0 +1,73 @@
+import { useMemo, useContext } from 'react';
+
+import { UserSubsidyContext } from '../../enterprise-user-subsidy/UserSubsidy';
+import { CourseContext } from '../CourseContextProvider';
+import {
+  findOfferForCourse, hasCourseStarted, findUserEnrollmentForCourse,
+} from '../data/utils';
+
+/**
+ * Enrollment data needed for enroll logic
+ * @typedef {Object} EnrollmentData
+ * @property {boolean} isUserEnrolled
+ * @property {boolean} isCourseStarted
+ * @property {boolean} isEnrollable
+ * @property {object} userEnrollment
+ */
+
+/**
+ *
+ * Extracts info needed for enroll button logic from CourseContext, for a single course.
+ * Which course?
+ *   `activeCourseRun`, `userEnrollments` are looked up in the CourseContext's `state`
+ *
+ * @returns {EnrollmentData} enrolldata data for use in enrollment logic
+ */
+export function useEnrollData() {
```

Comment:
got it

Reply:
hmm I now see that most functions use `export function` (40 of them) in this repo, there are a few arrow functions too, for smaller functions one liners or `isNull` and that kind of stuff. Looks like we are largely using `export const` to export constant values, but also some arrow functions.

so shall we leave these be or do you prefer the arrow ones? I guess I can see both styles working for these functions esp since `this` is not being used

Haha can't decide yet

Reply:
Yeah, I don't think we need to be nitpicky on how we create functions unless there's explicit reason to be. I tend to prefer using arrow functions these days as well though.